ladybug_geometry/geometry2d/polygon.py

```diff
@@ -717,6 +717,94 @@ def _are_clockwise(vertices):
             _a += vertices[i - 1].x * pt.y - vertices[i - 1].y * pt.x
         return _a < 0
 
+    @staticmethod
+    def _is_close(a, b, rel_tol=1e-9, abs_tol=0.0):
+        """
+        Implementation of PEP0485 to enable python version prior to 3.5 to perform floating-point comparisons.
+         See: https://www.python.org/dev/peps/pep-0485/#proposed-implementation
+
+        Args:
+            a and b: are the two values to be tested to relative closeness.
+            rel_tol: is the relative tolerance -- it is the amount of error allowed, relative
+                     to the larger absolute value of a or b.  Must be greater than 0.0.
+            abs_tol: is a minimum absolute tolerance level -- useful for comparisons near zero.
+
+        Returns:
+            True if the two numbers a and b are relatively close.  False otherwise.
+        """
+        return abs(a - b) <= max(rel_tol * max(abs(a), abs(b)), abs_tol)
+
+    @staticmethod
+    def _intersect_polygon_segments(polygon1, polygon2, tolerance=0.0):
+        """Check two adjacent polygons to see if one contains a vertex along an edge segment
+        of the other within the given tolerance. Creates a co-located vertex at that point,
+        partitioning the edge segment into two edge segments.  Point ordering is preserved.
+
+        Args:
+            polygon1: First polygon to check.
+            polygon2: Second polygon to check.
+            tolerance: Distance within which two points are considered to be co-located.
+
+        Returns:
+            Two polygon objects with extra vertices inserted if necessary.
+        """
+        polygon1_updates = []
+        polygon2_updates = []
+
+        # Bounding rectangle check
+        polygon1_width = polygon1.max.x - polygon1.min.x
+        polygon1_height = polygon1.max.y - polygon1.min.y
+        polygon2_width = polygon2.max.x - polygon2.min.x
+        polygon2_height = polygon2.max.y - polygon2.min.y
+        if (abs(polygon1.min.x - polygon2.min.x) * 2 <= (polygon1_width + polygon2_width + tolerance)) and \
+                (abs(polygon1.min.y - polygon2.min.y) * 2 <= (polygon1_height + polygon2_height + tolerance)):
+            pass
+        else:
+            return polygon1, polygon2
+
+        for i1, seg1 in enumerate(polygon1.segments):
+            for i2, seg2 in enumerate(polygon2.segments):
+                x = closest_point2d_on_line2d(seg2.p1, seg1)
+                if all([p.distance_to_point(x) > tolerance for p in polygon1.vertices]) and\
```

Comment:
I hate to be the PEP8 👮 but I think there should be a whitespace after `and`.

Reply:
and if I'm not mistaken you don't need the brackets. Using a generator should work just fine and will be more efficient:

```suggestion
                if all(p.distance_to_point(x) > tolerance for p in polygon1.vertices) and \
```

Reply:
Ah, good to know @mostaphaRoudsari . I think I should change this in some of my code as well.

Reply:
@chriswmackey I think you're correct on that whitespace (although PyCharm didn't flag it as a PEP8 violation, oddly).  I've added it in anyway for consistency.
@mostaphaRoudsari that's a good catch!  The lazy-evaluation of generators is definitely more efficient.  Changes made.